api/src/syscall/signal.rs

```diff
@@ -291,7 +293,9 @@ pub fn sys_rt_sigsuspend(
         Poll::Pending
     }));
 
-    Ok(0)
+    // Return -EINTR to match Linux behavior
+    // sigsuspend always returns -1 with errno=EINTR after handling a signal
```

Comment:
To nitpick a bit, `returns -1 with errno=EINTR` refers to the behavior of libc functions, right? For syscalls, `-EINTR` is more appropriate.


```suggestion
    // sigsuspend always returns -EINTR
```